compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -288,13 +287,14 @@ fn late_arg_as_bound_arg<'tcx>(
 ) -> ty::BoundVariableKind {
     match arg {
         ResolvedArg::LateBound(_, _, def_id) => {
-            let name = tcx.hir().name(tcx.local_def_id_to_hir_id(def_id.expect_local()));
+            let def_id = def_id.to_def_id();
+            let name = tcx.item_name(def_id);
```

Comment:
I've got a vague recollection that there's something subtly different between `item_name` and `hir::Map::name`... Something around elided lifetimes... Any idea?